File: src/main/java/com/dude/dms/ui/builder/docs/DocCreateDialogBuilder.java
package com.dude.dms.ui.builder.docs;

import com.dude.dms.ui.EntityCreateListener;
import com.dude.dms.ui.components.dialogs.DocCreateDialog;

public final class DocCreateDialogBuilder {

    private EntityCreateListener eventListener;

    DocCreateDialogBuilder() {
    }

    public DocCreateDialogBuilder withEventListener(EntityCreateListener eventListener) {
        this.eventListener = eventListener;
        return this;
    }

    public DocCreateDialog build() {
        DocCreateDialog dialog = new DocCreateDialog();
        dialog.setCreateListener(eventListener);
        return dialog;
    }
}

File: src/main/java/com/dude/dms/ui/builder/docs/DocCreateDialogBuilder.kt
package com.dude.dms.ui.builder.docs

import com.dude.dms.brain.polling.PollingService
import com.dude.dms.ui.builder.Builder
import com.dude.dms.ui.components.dialogs.DocCreateDialog

class DocCreateDialogBuilder(private val pollingService: PollingService): Builder<DocCreateDialog> {

    override fun build() = DocCreateDialog(pollingService)
}